lib/src/hash_sink.dart

```diff
@@ -133,9 +141,11 @@ abstract class HashSink implements Sink<List<int>> {
     var lengthInBits = _lengthInBytes * bitsPerByte;
 
     // Add the full length of the input data as a 64-bit value at the end of the
-    // hash.
-    var offset = _pendingData.length;
-    _pendingData.addAll(Uint8List(8));
+    // hash. Note: we're only writing out 64 bits, so skip ahead 8 if the
+    // signature is 128-bit.
+    final offset = _pendingData.length + (signatureBytes - 8);
```

Comment:
What if signatureBytes is less than 8?

Should probably document that it mustn't be, and check with an assert here.

Reply:
It can only be 8 or 16 according to SHA-2. I'll add an assert and document!